cloudslang-api-commons/src/test/java/io/cloudslang/lang/commons/services/impl/SlangCompilationServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * (c) Copyright 2016 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package io.cloudslang.lang.commons.services.impl;
+
+import io.cloudslang.lang.api.Slang;
+import io.cloudslang.lang.commons.services.api.CompilationHelper;
+import io.cloudslang.lang.commons.services.api.SlangCompilationService;
+import io.cloudslang.lang.compiler.SlangSource;
+import org.fusesource.jansi.Ansi;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = SlangCompilationServiceImplTest.Config.class)
+public class SlangCompilationServiceImplTest {
+
+    @Autowired
+    private SlangCompilationService slangCompilationService;
+
+    @Autowired
+    private CompilationHelper compilationHelper;
+
+    @Autowired
+    private Slang slang;
+
+    @Test
+    public void testLoadInputsFromFile() throws URISyntaxException {
+        File folder = new File(getClass().getResource("/executables").toURI().getPath());
+        Collection<File> files = slangCompilationService.listSlangFiles(folder, true);
+        assertEquals(4, files.size());
+    }
+
+    @Test
+    public void testCompileFoldersCleanup() throws Exception {
+        List<String> folders = new ArrayList<>();
+        folders.add(getClass().getResource("/executables").toURI().getPath());
+        slangCompilationService.compileFolders(folders, compilationHelper);
+
+        InOrder inOrderHelper = inOrder(compilationHelper);
+        inOrderHelper.verify(compilationHelper, times(4)).onEveryFile(any(Ansi.Color.class), anyString());
+        inOrderHelper.verify(compilationHelper).onCompilationFinish();
+        inOrderHelper.verifyNoMoreInteractions();
+        InOrder inOrder = inOrder(slang);
+        inOrder.verify(slang, times(4)).compileSource(
```

Comment:
it is a little better to actually have inOrder.verify with normal times(1) and specify the exact `SlangSource` for each such call of `compileSource ` method